File: Leetcode/cp1968_array_with_elements_not_equal_to_average_of_neighbors/Solution.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int i = 1;
        while (i < nums.length-1) {
            int sum = nums[i-1] + nums[i+1];
            if (sum % 2 == 0 && nums[i] == Math.floor(sum / 2)) {
                swap(nums, i, i + 1);
                if (i > 1) {
                    i--;
                }
            } else {
                i++;
            }
        }
        return nums;
    }
    
    public void swap(int[] array, int i, int j) {
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }
}

File: Leetcode/cp1968_array_with_elements_not_equal_to_average_of_neighbors/Solution.py
def swap(nums, i, j):
    nums[i], nums[j] = nums[j], nums[i]

class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        i=1
        while i < n-1:
            sum = nums[i-1]+nums[i+1]
            if sum%2 == 0 and nums[i] == sum // 2:
                swap(nums, i, i+1)
                if i > 1:
                    i-=1
            else:
                i+=1
        return nums